### PR TITLE
Bump PHP-Parser

### LINKED ISSUE #1353: Installation failed: Conclusion: remove nikic/php-parser v4.10.2
```bash
composer require --dev infection/infection
```

```
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Conclusion: don't install infection/infection 0.18
    - Conclusion: remove nikic/php-parser v4.10.2
    - Installation request for infection/infection ^0.18.0 -> satisfiable by infection/infection[0.18, 0.18.x-dev].
    - Conclusion: don't install nikic/php-parser v4.10.2
    - infection/infection 0.18.x-dev requires nikic/php-parser ^4.9 <4.10 -> satisfiable by nikic/php-parser[4.9.x-dev, v4.9.0, v4.9.1].
    - Can only install one of: nikic/php-parser[4.9.x-dev, v4.10.2].
    - Can only install one of: nikic/php-parser[v4.9.0, v4.10.2].
    - Can only install one of: nikic/php-parser[v4.9.1, v4.10.2].
    - Installation request for nikic/php-parser (locked at v4.10.2) -> satisfiable by nikic/php-parser[v4.10.2].


Installation failed, reverting ./composer.json to its original content.
```

Workaround:

```bash
composer require nikic/php-parser:4.9
```

And retry:

```bash
composer require --dev infection/infection
```

### PR DESCRIPTION
**Pull request by sanmai**
This PR:

- [x] Updates PHP-Parser to use the latest version
- [x] Fixes #1353

~~PHP-Parser had an issue, but apparently it's gone in newer versions.~~ It is still there:

```
PHP Fatal error:  Uncaught Error: Undefined constant '_HumbugBox2855bd3aa92c\T_NAME_QUALIFIED' 
in phar:///infection.phar/vendor/nikic/php-parser/lib/PhpParser/Lexer.php:301
```

### CODE REVIEW

**Comment by maks-rafalko** [comment]
Did I get it right that we can't merge it? (because of that issue)

**Comment by sanmai** [comment]
No, we can't. Phar build is broken.

**Comment by sanmai** [comment]
I think I'll leave thing handing just on the off chance @theofidry can know what's up here.

**Comment by sanmai** [comment]
@maks-rafalko I can build it just fine locally.

File: scoper.inc.php [hunk 1]
```diff
@@ -37,6 +37,13 @@
     'whitelist' => [
         \Composer\Autoload\ClassLoader::class,
         'Safe\*',
+        // PHP 8.0
+        'T_NAME_QUALIFIED',
+        'T_NAME_FULLY_QUALIFIED',
+        'T_NAME_RELATIVE',
+        'T_MATCH',
+        'T_NULLSAFE_OBJECT_OPERATOR',
+        'T_ATTRIBUTE',
```
Comment (sanmai) [inline scoper.inc.php:46]: Taken from here: https://github.com/nikic/PHP-Parser/blob/1d1bc8a36459d67acef508b952dad097960b110e/lib/PhpParser/Lexer.php#L413

**Review by maks-rafalko — APPROVED** [review]
:+1:

**Comment by sanmai** [comment]
This should be not very hard to backport.

File: scoper.inc.php (hunk 1 above)
Comment (theofidry) [reply scoper.inc.php:46]: >Could this be added to https://github.com/humbug/php-scoper/blob/master/src/Reflector.php#L64?

Comment (sanmai) [reply scoper.inc.php:46]: Absolutely. I thought there must be something like this, but couldn't find this straight away.

**Pull request merged**